### PR TITLE
Sort all deps in Cargo.tomls alphabetically

### PR DESCRIPTION
**Pull request by nrc**
###  What have you changed?

Sort dependencies in Cargo.tomls alphabetically

PTAL @Hoverbear @brson 

###  What is the type of the changes?

- Engineering (engineering change which doesn't change any feature or fix any issue)

### CODE REVIEW

**Comment by siddontang** [comment]
do we have a tool to do this automatically? or a format tool?

**Comment by nrc** [comment]
> do we have a tool to do this automatically? or a format tool?

Yes, but it is not quite ready to use in CI

File: Cargo.toml [hunk 1]
```diff
@@ -38,79 +38,79 @@ protobuf-codec = [
...
+slog = { version = "2.3", features = ["max_level_trace", "release_max_level_debug"] }
+slog-async = { version = "2.3", default-features = false }
+slog-global = { version = "0.1", git = "https://github.com/breeswish/slog-global.git", rev = "0e23a5baff302a9d7bccd85f8f31e43339c2f2c1" }
+slog-term = "2.4"
+slog_derive = "0.1"
+spin = "0.5"
+sst_importer = { path = "components/sst_importer" }
+sys-info = "0.5.7"
+tempfile = "3.0"
+test_util = { path = "components/test_util" }
```
Comment (breeswish) [inline Cargo.toml]: I prefer to separate project workspace dependencies from 3rd party dependencies to be more clear. The rest is fine!

Comment (nrc) [reply Cargo.toml]: Unfortunately the tooling doesn't support that. It is maybe something we can implement later.

I think this also may run into the same problem as with imports, that there are lots of grey areas with things that could be in the workspace, e.g., git deps, things like kvproto which might be versioned, other PingCAP crates, ...

**Comment by breeswish** [comment]
I hope we can set up the CI checking & local format tool first, always ensure the dependency sorting. Otherwise this sorting is very likely to be broken in the very near future..

**Comment by nrc** [comment]
> I hope we can set up the CI checking & local format tool first, always ensure the dependency sorting. Otherwise this sorting is very likely to be broken in the very near future..

I think it is OK - of course it would be better to have the tooling complete, but I think it is better to land these changes without it than to wait. There is an immediate benefit in making it easier to find deps in the manifest files, and it is also better (or at least no worse) to have them partially sorted than not sorted at all. (In other words it is fine to land this and for the deps to get out of order later).

**Pull request merged**